dashboard/app/models/blockly.rb

```diff
@@ -119,6 +119,21 @@ def pretty_block(block_name)
     self.class.pretty_print(xml_string)
   end
 
+  def self.count_xml_blocks(xml_string)
+    unless xml_string.blank?
+      xml = Nokogiri::XML(xml_string, &:noblanks)
+      # The structure of the XML will be
+      # <document>
+      #   <xml>
+      #     ... blocks ...
+      # So the blocks will be the children of the first child of the document
+      if xml.children && xml.children.first && xml.children.first.children
```

Comment:
you could one-line this as `return xml.try(:children).try(:first).try(:children).try(:length) || 0`